scripts/json2synapse.py

```diff
@@ -0,0 +1,217 @@
+
+    flatten_vals.append(empty_vals)
+
+    return flatten_vals
+
+
+def rightJoin(df1, df2, key, suffixes):
+    # rows in df2 that are not in df1 based on unique keys (can be a list) 
+    df3 = pandas.merge(df1, df2, on=key, how='right', indicator='Exist', suffixes=suffixes)
+    df3['Exist'] = numpy.where(df3.Exist == 'both', True, False)
+    return df3
+
+
+def outerJoin(df1, df2):
+    # rows in df1 that are not in df2 
+    df3 = pandas.merge(df1, df2, how='outer', indicator=True).query('_merge == "left_only"').drop(['_merge'], axis=1)
+    return df3
+
+
+def updateTable(tableSynId, newTable, key, delta=False, whereClause=False):
+    """
+    A table may have one-many key(s) in a table schema predefined by the user
+    There are three cases where changes can be done to a table:
+
+        Case 1: The unique keys have been changed but the remaining schema is the same
+        Case 2: The schema has been changed but the keys are the same
+        Case 3: New rows that have been added to table
+
+    :param tableSynId: the synapse table Id
+    :param newTable: synapseAnnotations normalized and melted json files into a new dataframe
+    :param key: the unique identifiers of the schema in our case the tuple (key-value)
+    :param delta: if previous version of changes should not be deleted
+    :param whereClause: synapse table query where clause that follows a SQL language syntax
+    :return: None and table on synapse identified by tableSynId should be updated
+    """
+
+    currentTable = syn.tableQuery("SELECT * FROM %s" % tableSynId)
+    currentTable = currentTable.asDataFrame()
+
+    if sum(pandas.Series(key).isin(currentTable.columns)) == 0:
+        sys.exit("All of the unique keys (key | value) in schema don't exist")
+    elif sum(pandas.Series(key).isin(currentTable.columns)) == 1:
+        sys.exit("One of the unique keys (key | value) in schema don't exist")
+    else:
+        logging.info("Unique keys in schema exist and pursuing")
+
+    # Test cases: change a non-key column cell (valuesDescription) and change a unique key cell (value)
+    # newTable.iloc[3, newTable.columns.get_loc('valuesDescription')] = 'exome sequencing Nasim'
+    # newTable.iloc[1, newTable.columns.get_loc('value')] = 'Nasim'
+
+    # standardize na/NAN in data by replacing it with an empty string for match comparisons
+    currentTable = currentTable.fillna("")
+    newTable = newTable.fillna("")
+
+    # Match schema in current and new tables
+    schema = currentTable.columns
+    newTable = newTable[currentTable.columns]
+
+    # Keep the ROW_ID of the current Table
+    currentTable['index'] = currentTable.index
+
+    ## Case 1: find the rows where the unique keys have been changed but the remaining schema is the same
+    resultDF = rightJoin(df1=currentTable, df2=newTable, key=key, suffixes=('_cur', '_new'))
+
+    newCol = key + [s + '_new' for s in [col for col in newTable.columns if col not in key]]
```

Comment:
nope. Its the new table columns that aren't `unique key` columns concatenated by the string '_new'
```
s + '_new' for s in list(set(df2.columns) - set(['key1', 'key2', 'key3']))
```